BrainPortal/lib/boutiques_input_value_fixer.rb

```diff
@@ -0,0 +1,192 @@
+    # the major use case are Flags, but also may be useful to address params with 'default'
+
+    descriptor_dup = descriptor.dup
+    todelete = fixed_values.keys.select do |i_id|  # this variables are flagged to be removed rather than assigned value
+                                                   # in the spec, so they will be treated slightly different
+      input = descriptor_dup.input_by_id(i_id)
+      value = fixed_values[i_id]
+      value.nil? || (input.type == 'Flag') &&  (value.presence.to_s.strip =~ /no|0|nil|none|null|false/i || value.blank?)
+
+    end
+
+    # generally speaking, boutiques input groups can have three different constraints,
+    # here we address mutually exclusion group only, which is the only one present in dynamic GUI (rest are evaluated
+    # after submission of parameter).
+
+    descriptor_dup.groups.each do |g| # filter groups, relax restriction to ensure that form can still be submitted
+      members = g.members - fixed_values.keys
+      # disable a mutualy exclusive group if its param assigned fixed value by this modifier
+      # if one simply deletes the fixed param,
+      if g.try(:mutually_exclusive) && members.length != g.members.length # params can be mutually exclusive e.g. --use-min-mem vs --mem-mb
+        if (fixed_values.keys & g.members - todelete).present? # at least some group members are actually assigned vals rather than deleted
+          g.mutually_exclusive = false
+          if g.try(:one_is_required).present?
+            g.one_is_required = false
+          end
+          block_inputs(descriptor_dup, members - fixed_values.keys)
+        end
+      end
+
+      # all-or-none is not reflected in dynamic gui, perhaps address (uncomment) once gui is fixed
+
+      # removes  'one-is-required' or disables group when one or more element fixed, e.g.
+      # if g.all_or_none && members.length != g.members.length
+      #   # if (g.members & todelete).present?
+      #   #   # todo delete all member inputs, or disable by injecting pairwise required/disable dependencies
+      #   # end
+      #   if (fixed_values.keys & g.members - todelete).present? # if one is set, rest should be to
+      #     g.members.each do |i_id|
+      #       begin
+      #         input = descriptor.input_by_id(i_id)
+      #       rescue CbrainError  # if descriptor was already processed
+      #         next
+      #       end
+      #       input.optional = false
+      #     end
+      #   end
+      # end
+
+      # presently one-is-required is checked only statically, no dinamic GUI support
+      # removes one-is-required flag if one element fixed
+      # if g.one_is_required && members.length != g.members.length
+      #   if (fixed_values.keys & g.members - todelete).present?
+      #     g.one_is_required == false
+      #   end
+      # end
+
+      g.members = members
+    end
+    descriptor_dup.groups = descriptor_dup.groups.select {|g| g.members.present? } # delete empty group
```

Comment:
You're already working on a full dup of the descriptor, so use the mutator methods instead, that will avoid a lot of copying:

```ruby
descrtiptor_dup.groups.reject! { |g| condition }
```

That also makes the code way easier to read.